cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -0,0 +1,379 @@
+    }
+  }
+
+  private static void parseArgs(String args[]) {
+    for (int i = 0; i < args.length;) {
+      if (args[i].charAt(0) == '-') {
+        switch (args[i].charAt(1)) {
+          case 'm':
+            masterClusterID = args[i + 1];
+            i += 2;
+            break;
+          case 'i':
+            inputStatusFileName = args[i + 1];
+            i += 2;
+            break;
+          case 'o':
+            outputStatusFileName = args[i + 1];
+            i += 2;
+            break;
+          case 'd':
+            // Debug dump
+            debugDump = true;
+            i += 1;
+            break;
+        }
+      } else {
+        return;
+      }
+    }
+  }
+
+  private static void showHelp() {
+    System.out.println("\nTool to check Replication Status.");
+    System.out.println("Run this tool on the Master cluster:");
+    System.out.println("#ReplicationStatusTool -m <master cluster id> -o <output filepath>");
+    System.out.println("Then run on the Slave Cluster:");
+    System.out.println("#ReplicationStatusTool -m <master cluster id> -i <file copied from Master>\n");
+    return;
+  }
+
+  private static void processMasterCluster() throws IOException {
+    //Get checksum from Snapshot files
+    masterChecksum = getClusterChecksum();
+    masterShutdownTime = getShutdownTime();
+
+    if (masterShutdownTime == 0L) {
+      System.out.println("CDAP Shutdown time not available. Please run after CDAP has been shut down.");
+      return;
+    }
+
+    //Scan the table and write to output file
+    writeMasterStatusFile(getMapFromTable(ReplicationConstants.ReplicationStatusTool.WRITE_TIME), outputStatusFileName);
+    System.out.println("Copy the file " + outputStatusFileName + " to the Slave Cluster and run tool there.");
+  }
+
+  private static void processSlaveCluster() throws Exception {
+    slaveClusterID = localClusterID;
+    Map<String, Long> masterTimeMap = readMasterStatusFile(inputStatusFileName);
+
+    if (masterShutdownTime == 0L) {
+      System.out.println("Could not read CDAP Shutdown Time from input file."
+                           + " Please make sure CDAP has been shutdown on the Master and rerun the tool on Master.");
+      return;
+    }
+    if (masterChecksum == null) {
+      System.out.println("Could not read Master Checksum from input file.");
+    }
+    if (masterTimeMap == null || masterTimeMap.size() == 0) {
+      LOG.info("No region information found in the input file");
+    }
+
+    checkHBaseReplicationComplete(masterTimeMap);
+    //checkSnapshotReplicationComplete(masterChecksum);
+  }
+
+  private static void checkHBaseReplicationComplete(Map<String, Long> masterTimeMap) throws Exception {
+    // Get replicate Time from table for all regions
+    Map<String, Long> slaveTimeMap = getMapFromTable(ReplicationConstants.ReplicationStatusTool.REPLICATE_TIME);
+
+    //Check all regions are accounted on both clusters
+    if (masterTimeMap.size() != slaveTimeMap.size()) {
```

Comment:
If this is false, the immediate follow up question will be to ask which regions are missing. It would be more informative to compare the actual regions and print any differences. You can use guava's Sets.difference() to help.